src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -1504,6 +1504,23 @@ void Leader::HandleTimer(void)
     }
 }
 
+void Leader::RemoveStaleChildEntries(void)
+{
+    Iterator iterator = kIteratorInit;
+    uint16_t rloc16;
+
+    while (GetNextServer(iterator, rloc16) == OT_ERROR_NONE)
```

Comment:
LGTM. 

I think moving this to `network_data_leader_ftd` is great. With this change, later we can remove the `GetNextServer()` and have a direct implementation of going through all the entries here instead.